uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/WorkDayTest.php

```diff
@@ -156,16 +159,38 @@ public function testCannotHaveMoreThanOfDayOfTheWeekForEachWeek() {
   }
 
   /**
-   * @dataProvider numberOfHoursDataProvider
+   * @dataProvider validNumberOfHoursDataProvider
    */
-  public function testNumberOfHoursShouldBeCalculatedFromTimesAndBreak($timeFrom, $timeTo, $break, $expectedNumberOfHours) {
-    $entity = $this->createBasicWorkDay([
-      'time_from' => $timeFrom,
-      'time_to' => $timeTo,
-      'break' => $break
-    ]);
+  public function testNumberOfHours($numberOfHours) {
+    $entity = $this->createBasicWorkDay(['number_of_hours' => $numberOfHours]);
+  }
 
-    $this->assertEquals($expectedNumberOfHours, $entity->number_of_hours);
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException
+   * @expectedExceptionMessage Number of Hours should be a valid number
+   *
+   * @dataProvider invalidFormatNumberOfHoursDataProvider
+   */
+  public function testNumberOfHoursMustBeANumber ($numberOfHours) {
+    $entity = $this->createBasicWorkDay(['number_of_hours' => $numberOfHours]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException
+   * @expectedExceptionMessage Number of Hours should be between 0 and 24
+   *
+   * @dataProvider outOfBoundsNumberOfHoursDataProvider
+   */
+  public function testNumberOfHoursMustBeBetween0And24 ($numberOfHours) {
+    $entity = $this->createBasicWorkDay(['number_of_hours' => $numberOfHours]);
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidWorkDayException
+   * @expectedExceptionMessage Number of Hours should be divisible by 0.25
+   */
+  public function testNumberOfHoursMustBeDivisableBy15Minutes () {
```

Comment:
Maybe have a data provider for this, so we can test this with more options?

Reply:
Done.